### PR TITLE
fix no registered selector for terminal quick fix ID

### LINKED ISSUE #168154: Error: No registered selector for ID: ms-vscode.npm-command
Running out of sources I see this:

```
[66219:1206/102201.181288:INFO:CONSOLE(216)] "%c  ERR color: #f33 No registered selector for ID: ms-vscode.npm-command: Error: No registered selector for ID: ms-vscode.npm-command
    at TerminalQuickFixService.registerQuickFixProvider (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/contrib/terminal/browser/terminalQuickFixService.js:26:23)
    at MainThreadTerminalService.$registerQuickFixProvider (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/api/browser/mainThreadTerminalService.js:205:75)
    at RPCProtocol._doInvokeHandler (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/services/extensions/common/rpcProtocol.js:381:27)
    at RPCProtocol._invokeHandler (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/services/extensions/common/rpcProtocol.js:366:45)
    at RPCProtocol._receiveRequest (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/services/extensions/common/rpcProtocol.js:307:32)
    at RPCProtocol._receiveOneMessage (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/services/extensions/common/rpcProtocol.js:234:26)
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/workbench/services/extensions/common/rpcProtocol.js:115:52
    at Listener.invoke (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:514:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:688:38)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:653:37)
    at BufferedEmitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/common/ipc.net.js:480:35)
    at PersistentProtocol._receiveMessage (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/common/ipc.net.js:746:45)
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/common/ipc.net.js:627:83
    at Listener.invoke (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:514:27)
    at PrivateEventDeliveryQueue.deliver (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:688:38)
    at Emitter.fire (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/common/event.js:653:37)
    at ProtocolReader.acceptChunk (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/common/ipc.net.js:271:37)
    at vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/common/ipc.net.js:238:61
    at Socket.listener (vscode-file://vscode-app/Users/bpasero/Development/Microsoft/vscode/out/vs/base/parts/ipc/node/ipc.net.js:63:17)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
```

### PR DESCRIPTION
**Pull request by meganrogge**
fix #168154

could repro this when an OSS window opened to the VS Code folder

The issue was that the `quickFixAddon` was formerly calling `quickFixService` `registerCommandSelector` with the `terminalContributionService.quickFix` and that was happening too late. 

We avoid that delay now by registering the command selectors in the `terminalQuickFixService` so that they're available before `registerQuickFixProvider` is ever called

### CODE REVIEW

**Comment by meganrogge** [comment]
my plan is to move the `TerminalContributionService` into `TerminalQuickFixService`. 

Once command selectors have been registered, I'll allow registering of quick fix providers - caching the requests until such time

**Comment by meganrogge** [comment]
oh perhaps this is the issue

https://github.com/microsoft/vscode/blob/d733e6d696bad258db2b42fe5f4331b3a456acce/src/vs/workbench/contrib/terminal/common/terminalExtensionPoints.contribution.ts#L9

File: src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts [hunk 1]
```diff
@@ -193,7 +188,7 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 			await this._extensionService.activateByEvent(`onTerminalQuickFixRequest:${id}`);
 			const provider = this._quickFixService.providers.get(id);
 			if (!provider) {
-				this._logService.warn('No provider when trying to resolve terminal quick fix for provider: ', id);
+				this._logService.trace('No provider when trying to resolve terminal quick fix for provider: ', id);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts:191]: Warn is better here isn't it as it's unexpected?

File: src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts [hunk 2]
```diff
@@ -101,6 +96,7 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 			});
 		}
 		this._quickFixService.onDidRegisterProvider(result => this.registerCommandFinishedListener(convertToQuickFixOptions(result)));
+		this._quickFixService.onDidRegisterCommandSelector(selector => this.registerCommandSelector(selector));
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts:99]: You might have a race condition here, I think you also need to iterate over any command selectors that exist at the time `TerminalQuickFixAddon` is created?

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**